airflow/models/dagrun.py

```diff
@@ -311,32 +312,41 @@ def update_state(self, session=None):
 
         leaf_tis = [ti for ti in tis if ti.task_id in {t.task_id for t in dag.leaves}]
 
-        # if all roots finished and at least one failed, the run failed
-        if not unfinished_tasks and any(
-            leaf_ti.state in {State.FAILED, State.UPSTREAM_FAILED} for leaf_ti in leaf_tis
-        ):
-            self.log.info('Marking run %s failed', self)
-            self.set_state(State.FAILED)
-            dag.handle_callback(self, success=False, reason='task_failure',
-                                session=session)
-
-        # if all leafs succeeded and no unfinished tasks, the run succeeded
-        elif not unfinished_tasks and all(
-            leaf_ti.state in {State.SUCCESS, State.SKIPPED} for leaf_ti in leaf_tis
-        ):
-            self.log.info('Marking run %s successful', self)
-            self.set_state(State.SUCCESS)
-            dag.handle_callback(self, success=True, reason='success', session=session)
-
-        # if *all tasks* are deadlocked, the run failed
-        elif (unfinished_tasks and none_depends_on_past and
-              none_task_concurrency and no_dependencies_met):
-            self.log.info('Deadlock; marking run %s failed', self)
-            self.set_state(State.FAILED)
-            dag.handle_callback(self, success=False, reason='all_tasks_deadlocked',
-                                session=session)
-
-        # finally, if the roots aren't done, the dag is still running
+        if conf.getboolean('scheduler', 'REMOVED_TASKS_LEAD_TO_DAGRUN_FAILURE', fallback=False):
```

Comment:
Hi. Can you add some documentation?  We do not yet have a document describing the configuration options of the scheduler. Would you like to create it? If not, that's fine and you can also create a new document in the howto directory.

Reply:
@mik-laj should we make this a part of the airflow site? I want to start writing better docs for configurations and am not sure where would be the best place to put it.

Reply:
All documentation should be stored here: https://github.com/apache/airflow/tree/master/docs

"airflow-site" repository contains only the landing pages and generated HTML files for archival purposes.

Reply:
The documentation structure is a topic that is not taken seriously yet, as there is a lack of documentation for a large part of the options. If you have no idea where to place the documentation, you can create a new document in the `docs/howto` directory.  You can also call me and I will gladly give you tips in a specific situation.

Reply:
docs/scheduler.rst updated. @mik-laj  pls merge